File: p/shaker.py
class shaker_sort():
    def __new__(cls,arr:list)->list:
        left=0
        right = len(arr)-1
        while left<right:
        #left to right
            for i in range(left,right):
                if arr[i]>arr[i+1]:
                    arr[i],arr[i+1]=arr[i+1],arr[i]
            right-=1
        #right to left
            for j in range(right,left,-1):
                if arr[j]<arr[j-1]:
                    arr[j],arr[j-1]=arr[j-1],arr[j]
            left+=1
        return arr

File: r/shaker.r
shaker_sort<-function(arr){
  l<-1
  r<-length(arr)
  while (l<r){
    for(i in l:r){
      if(arr[i]>arr[i+1]){
        t <- arr[i]
        arr[i]<- arr[i+1]
        arr[i+1]<-t
      }
    }
    r<-r-1
    for(j in r:l+1){
      if(arr[j]<arr[j-1]){
        t<-arr[j]
        arr[j]<-arr[j-1]
        arr[j-1]<-t
      }
    }
    l<-l+1}
  return (arr)
}
